build/generate/functionTypes.go

```diff
@@ -0,0 +1,232 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+
+	"golang.org/x/text/cases"
+	"golang.org/x/text/language"
+	"gopkg.in/yaml.v3"
+)
+
+func join(parts ...string) string {
+	return strings.Join(parts, string(os.PathSeparator))
+}
+
+func fixRuns(s string) string {
+	lines := strings.Split(s, "\n")
+	var out []string
+	for _, line := range lines {
+		if len(line) == 0 {
+			if len(out) > 0 && len(out[len(out)-1]) == 0 {
+				continue
+			}
+		}
+		out = append(out, line)
+	}
+	return strings.Join(out, "\n")
+}
+
+var delimiterRegex = regexp.MustCompile(`(?m)^---\n`)
+func parseFrontMatter(content string) (map[string]interface{}, string, error) {
+	delimiterIndices := delimiterRegex.FindAllStringIndex(content, 2)
+	startIndex := delimiterIndices[0][0]
+	endIndex := delimiterIndices[1][0]
+	yamlString := content[startIndex+4:endIndex]
+	var frontMatter map[string]interface{}
+	err := yaml.Unmarshal([]byte(yamlString), &frontMatter)
+	if err != nil {
+		return nil, "", err
+	}
+	return frontMatter, content[endIndex+4:], nil
+}
+
+var returnTypes = map[string]string{
+	"domain": "DomainModifier",
+	"global": "void",
+	"record": "RecordModifier",
+}
+var paramTypeDefaults = map[string]string{
+	"name": "string",
+	"target": "string",
+	"value": "string",
+	"destination": "string",
+	"priority": "number",
+	"registrar": "string",
+	"source": "string",
+	"ttl": "Duration",
+	"modifiers...": "RecordModifier[]",
+}
+
+func generateFunctionTypes() (string, error) {
+	funcs := []Function{}
+
+	srcRoot := join("docs", "_functions")
+	types, err := os.ReadDir(srcRoot)
+	if err != nil {
+		return "", err
+	}
+	for _, t := range types {
+		if !t.IsDir() {
+			return "", errors.New("not a directory: " + join(srcRoot, t.Name()))
+		}
+		tPath := join(srcRoot, t.Name())
+		funcNames, err := os.ReadDir(tPath)
+		if err != nil {
+			return "", err
+		}
+
+		for _, f := range funcNames {
+			fPath := join(tPath, f.Name())
+			if f.IsDir() {
+				return "", errors.New("not a file: " + fPath)
+			}
+			// println("Processing", fPath)
+			content, err := os.ReadFile(fPath)
+			if err != nil {
+				return "", err
+			}
+			frontMatter, body, err := parseFrontMatter(string(content))
+			if err != nil {
+				return "", err
+			}
+			if frontMatter["ts_ignore"] == true {
+				continue
+			}
+
+			body = body + "\n"
+			body = strings.ReplaceAll(body, "{{site.github.url}}", "https://www.dnscontrol.org/")
```

Comment:
After @tlimoncelli has undone the Github custom domain configuration https://github.com/StackExchange/dnscontrol/issues/1843#issuecomment-1337828223, these (links) no longer work.